backend/azure/dtmb/dtmb.go

```diff
@@ -0,0 +1,531 @@
+		InstanceID: api.InstanceID(item.OrchestrationId),
+		NewEvent:   &genericInputWrapper,
+		Properties: map[string]interface{}{
+			"CompletionToken": item.GetCompletionToken(),
+			"ActivityName":    item.GetName(),
+		},
+	}
+	return ret, nil
+}
+
+// CompleteActivityWorkItem sends a message to the parent orchestration indicating activity completion.
+//
+// Returns [ErrWorkItemLockLost] if the work-item couldn't be completed due to a lock-lost conflict (e.g., split-brain).
+func (d *dtmb) CompleteActivityWorkItem(_ context.Context, item *backend.ActivityWorkItem) error {
+	bytes, err := backend.MarshalHistoryEvent(item.Result)
+	if err != nil {
+		return err
+	}
+	var failureDetails *dtmbprotos.FailureDetails = nil
+	if item.NewEvent.GetTaskFailed() != nil {
+		failureDetails = utils.ConvertTaskFailureDetails(item.NewEvent.GetTaskFailed().GetFailureDetails())
+	}
+
+	completionToken := item.Properties["CompletionToken"].(string)
+	activityName := item.Properties["ActivityName"].(string)
+
+	d.connectWorkerClientStream <- &dtmbprotos.ConnectWorkerClientMessage{
+		Message: &dtmbprotos.ConnectWorkerClientMessage_CompleteActivity{
+			CompleteActivity: &dtmbprotos.CompleteActivityMessage{
+				OrchestrationId: string(item.InstanceID),
+				Name:            activityName,    // what do I put here?
+				CompletionToken: completionToken, // what do I put here?
+				Result:          bytes,           // is this correct?
+				FailureDetails:  failureDetails,  // which failures go here?
+			},
+		},
+	}
+	return nil
+}
+
+// AbandonActivityWorkItem returns the work-item back to the queue without committing any other chances.
+//
+// This is called when an internal failure occurs during activity work-item processing.
+func (d *dtmb) AbandonActivityWorkItem(_ context.Context, item *backend.ActivityWorkItem) error {
+	completionToken := item.Properties["CompletionToken"].(string)
+	activityName := item.Properties["ActivityName"].(string)
+
+	d.connectWorkerClientStream <- &dtmbprotos.ConnectWorkerClientMessage{
+		Message: &dtmbprotos.ConnectWorkerClientMessage_AbandonActivity{
+			AbandonActivity: &dtmbprotos.AbandonActivityMessage{
+				OrchestrationId: string(item.InstanceID),
+				Name:            activityName,
+				CompletionToken: completionToken,
+			},
+		},
+	}
+	return nil
+}
+
+// PurgeOrchestrationState deletes all saved state for the specified orchestration instance.
+//
+// [api.ErrInstanceNotFound] is returned if the specified orchestration instance doesn't exist.
+// [api.ErrNotCompleted] is returned if the specified orchestration instance is still running.
+func (d *dtmb) PurgeOrchestrationState(ctx context.Context, id api.InstanceID) error {
+	_, err := d.clientClient.PurgeOrchestration(ctx, &dtmbprotos.PurgeOrchestrationRequest{
+		Request: &dtmbprotos.PurgeOrchestrationRequest_OrchestrationId{
+			OrchestrationId: string(id),
+		},
+		NonRecursive: false, // TODO: Should this be globally configurable?
```

Comment:
Ideally, we would always make this non-configurable. We can consider it later, though - I don't expect anyone will want to enable non-recursive.

Reply:
Sounds good.